### PR TITLE
Port airlock painter fixes and code cleanups

### PR DESCRIPTION
**Pull request by MNarath1**
## About The Pull Request

Ports https://github.com/yogstation13/Yogstation/pull/10870 for the most part
and Ports the assoc list for airlock painters from https://github.com/yogstation13/Yogstation/pull/10116.
This PR was made as a better solution then https://github.com/BeeStation/BeeStation-Hornet/pull/3649.
The Port was tweaked and tested by myself to ensure you can properly repaint external airlocks.

## Why It's Good For The Game

Fixes a cult airlock painter exploid and an icon bug with airlock painter

## Changelog
:cl: MNarath1, JamieD1, Redmoogle
fix: Repainted airlocks sometimes missing the moving parts, or having wrong animation
fix: Exploit allowing cult to disguise their airlocks with airlock painter
tweak: Disallowed repaint of large airlock and hatches
refactor: Changes code to use the assoc list for airlock repaint choosing
/:cl:

### CODE REVIEW

**Comment by utilbird** [comment]
It's generally good practice to add ported PR's author's name (and your own) beside the ":CL:"

**Comment by MNarath1** [comment]
I tried to define them directly as coauthors but yeah that works too

**Comment by MNarath1** [comment]
@TheFakeElon Ok i added them now there too wanted initially to use coauthor to add them but i think i messed that up as it obviously did not work too well

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -1251,80 +1252,35 @@
...
+
+	// reads from the airlock painter's `available paintjob` list. lets the player choose a paint option, or cancel painting
+	var/current_paintjob = input(user, "Please select a paintjob for this airlock.") as null|anything in sortList(painter.available_paint_jobs)
+	if(!current_paintjob) // if the user clicked cancel on the popup, return
+		return
 
-	var/paintjob = input(user, "Please select a paintjob for this airlock.") in sortList(optionlist)
-	if((!in_range(src, usr) && loc != usr) || !W.use_paint(user))
+	var/airlock_type = painter.available_paint_jobs["[current_paintjob]"] // get the airlock type path associated with the airlock name the user just chose
+	var/obj/machinery/door/airlock/airlock = airlock_type // we need to create a new instance of the airlock and assembly to read vars from them
```
Comment (park66665) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
	var/obj/machinery/door/airlock/airlock = airlock_type // we need to create a typed variable of the airlock and assembly to read initial values from them
```

**Review by park66665 — COMMENTED** [review]
You'll want `allow_repaint = FALSE` for `/obj/machinery/door/airlock/maintenance_hatch` and `/obj/machinery/door/airlock/abductor` if you want greater immersion.

External airlocks are in the grey zone: repainting to and fro external airlocks breaks immersion greatly, yet it has been allowed. But guess there's a reason why external airlocks can be painted.

The code is okay in general. If you want nothing to be changed, I will merge this PR at this state.

**Comment by MNarath1** [comment]
> You'll want `allow_repaint = FALSE` for `/obj/machinery/door/airlock/maintenance_hatch` and `/obj/machinery/door/airlock/abductor` if you want greater immersion.
> 
> External airlocks are in the grey zone: repainting to and fro external airlocks breaks immersion greatly, yet it has been allowed. But guess there's a reason why external airlocks can be painted.
> 
> The code is okay in general. If you want nothing to be changed, I will merge this PR at this state.

I will set allow_repaint = FALSE for abductor cause thats a unique airlock.

**Comment by MNarath1** [comment]
Ah screw it you are right it makes no sense that you can repaint the maint hatch or normal hatch.

**Review by park66665 — APPROVED** [review]
All clear. This PR is good to go.

**Pull request merged**